extensions/ccui/base-classes/UIScale9Sprite.js

```diff
@@ -167,36 +170,62 @@ ccui.Scale9Sprite = cc.Node.extend(/** @lends ccui.Scale9Sprite# */{
         locCenter.setScaleX(horizontalScale);
         locCenter.setScaleY(verticalScale);
 
-        var locLeft = this._left, locRight = this._right, locTop = this._top, locBottom = this._bottom;
-        var tempAP = cc.p(0, 0);
-        locBottomLeft.setAnchorPoint(tempAP);
-        locBottomRight.setAnchorPoint(tempAP);
-        locTopLeft.setAnchorPoint(tempAP);
-        locTopRight.setAnchorPoint(tempAP);
-        locLeft.setAnchorPoint(tempAP);
-        locRight.setAnchorPoint(tempAP);
-        locTop.setAnchorPoint(tempAP);
-        locBottom.setAnchorPoint(tempAP);
-        locCenter.setAnchorPoint(tempAP);
 
         // Position corners
-        locBottomLeft.setPosition(0, 0);
-        locBottomRight.setPosition(leftWidth + rescaledWidth, 0);
-        locTopLeft.setPosition(0, bottomHeight + rescaledHeight);
-        locTopRight.setPosition(leftWidth + rescaledWidth, bottomHeight + rescaledHeight);
+        if(locBottomLeft)
+        {
+            locBottomLeft.setAnchorPoint(cc.p(1,1));
```

Comment:
If the anchor point is fixed, it should be initialized as this and never be modified